packaging/tests/tas-restapi/src/main/java/org/alfresco/rest/repo/resource/Categories.java

```diff
@@ -0,0 +1,346 @@
+import org.alfresco.rest.repo.resource.general.MultipleResourcesCreator;
+import org.alfresco.rest.repo.resource.general.ResourceCreator;
+import org.alfresco.rest.repo.resource.general.ResourceModifier;
+import org.alfresco.rest.repo.resource.general.Specifier;
+import org.alfresco.utility.data.DataUser;
+import org.alfresco.utility.model.UserModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+/**
+ * Helper class simplifying things related with repository categories management.
+ */
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class Categories implements ResourceManager<RestCategoryModel, Specifier.CategoriesSpecifier, Modifier.CategoryModifier>
+{
+    private static final String CATEGORY_NAME_PREFIX = "category";
+    private static final String ROOT_CATEGORY_ID = "-root-";
+    public static final RestCategoryModel ROOT_CATEGORY = RestCategoryModel.builder().id(ROOT_CATEGORY_ID).create();
+
+    private final RestWrapper restClient;
+    private final UserModel user;
+
+    private final Map<String, RestCategoryModel> categoriesCache = new MultiKeyResourceMap<>(RestCategoryModel::getId, RestCategoryModel::getName);
+
+    public Categories(RestWrapper restClient, UserModel user)
+    {
+        this.restClient = restClient;
+        this.user = user;
+    }
+
+    @Autowired
+    public Categories(RestWrapper restClient, DataUser dataUser)
+    {
+        this(restClient, dataUser.getAdminUser());
+    }
+
+    @Override
+    public Specifier.CategoriesSpecifier add()
+    {
+        return (Specifier.CategoriesSpecifier) new RepoCategoryCreator(restClient, categoriesCache)
+            .underCategory(ROOT_CATEGORY).asUser(user);
+    }
+
+    @Override
+    public RestCategoryModel get(String key)
+    {
+        return categoriesCache.get(key);
+    }
+
+    @Override
+    public Modifier.CategoryModifier modify(RestCategoryModel category)
+    {
+        return new RepoCategoryModifier(restClient, category, categoriesCache).asUser(user);
+    }
+
+    @Override
+    public void delete(RestCategoryModel category)
+    {
+        new RepoCategoryModifier(restClient, category, categoriesCache).asUser(user).delete();
+    }
+
+    private static org.alfresco.rest.requests.Categories buildCategoryRestRequest(RestWrapper restClient, UserModel user, RestCategoryModel category)
+    {
+        return restClient.authenticateUser(user).withCoreAPI().usingCategory(category);
+    }
+
+    public static class RepoCategoryCreator
+        extends ResourceCreator<RestCategoryModel, Creator.CategoryCreator>
+        implements Creator.CategoryCreator, Specifier.CategoriesSpecifier
+    {
+        private final RestWrapper restClient;
+        private RestCategoryModel parent;
```

Comment:
Maybe this should default to ROOT_CATEGORY?

Reply:
Good point - fixed